juju/juju
# Clear the reboot flag in the worker before exiting.

If the juju-reboot hook command was executed by a charm that was running on the controller, the machine agent would attempt to connect to the apiserver after it had been shut down.

This change moves the "clear the reboot flag" work to the worker itself before it returns the error. Should there have been a problem stopping another of the services or listing machines for the container managers, the machine agent would have still been in a restart loop. This way clearing the flag out prior to returning the error, we also remove that restart loop problem.

Drive by fix for a uniter test race failure.

## QA steps

Grab the ubuntu-lite charm locally, and add an install hook with the following body:

```bash
#!/bin/bash
status-set maintenance rebooting
juju-reboot
```

Then execute the following:

```
juju bootstrap lxd test
juju deploy -m controller ../path/to/charm u --to 0
```

* check the logs of the controller machine to see that it restarted.
* juju show-status-log -m contgroller u/0 shows the charm started fine


## Bug reference

https://bugs.launchpad.net/juju/+bug/1810313
